### PR TITLE
Sidebar display fixes

### PR DESCRIPTION
**Pull request by thatbudakguy**
- Refactor sidebar component and combine secondary portions

Before:

Screenshot 2024-10-31 at 13 15 12

After:

Screenshot 2024-10-31 at 13 15 16

Part of #1437

### CODE REVIEW

File: app/views/relation/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<%- Settings.RELATIONSHIPS_SHOWN.each do |relationship_type, rel_type_info| %>
```
Comment (jcoyne) [inline app/views/relation/index.html.erb:1]: Where is this partial called from?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/components/document/sidebar_component.rb [hunk 2]
```diff
@@ -3,12 +3,33 @@
...
+      [
+        AlsoAvailableComponent.new(document:),
+        'relations_container',
+        Blacklight::Document::MoreLikeThisComponent.new(document:)
+      ]
+    end
+
+    # Determine if there's actually anything to show by rendering everything
+    def filter_components(components)
+      components.map { |component| render component, document: }.uniq.compact_blank
```
[the remark this answers is not in the document]
Comment (thatbudakguy) [reply app/components/document/sidebar_component.rb:27]: Not sure – I just extracted this from the template into a method. Doesn't seem like we could conceivably duplicate a component, so I'll remove it.

**Comment by thatbudakguy** [comment]
@jcoyne just so you're aware – `RelationsComponent` is a near-exact copy of Geoblacklight's `RelationsComponent`; all I wanted was to change a single CSS class in the template. If you want me to refactor it, I can, but that seems orthogonal to this PR.

**Comment by thatbudakguy** [comment]
I went ahead and removed the commit in this PR dealing with `RelationsComponent`; hopefully that makes it easier to review.

File: app/components/document/sidebar_component.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (thatbudakguy) [reply app/components/document/sidebar_component.rb]: Removed.

File: app/components/document/sidebar_component.rb [hunk 3]
```diff
@@ -3,12 +3,33 @@
...
         Geoblacklight::LoginLinkComponent.new(document:),
         Geoblacklight::StaticMapComponent.new(document:),
-        DownloadLinksComponent.new(document:)
+        DownloadLinksComponent.new(document:),
+        Blacklight::Document::ShowToolsComponent.new(document:)
       ]
     end
+
+    # Second section of the sidebar that displays below the main one
+    def secondary_components
```
[the remark this answers is not in the document]
Comment (thatbudakguy) [reply app/components/document/sidebar_component.rb:17]: Can I do that in Geoblacklight as part of a refactor to `RelationsComponent`? It seems like that's most of what you would like to see anyway. I can change the name of the method for now, if you like.

File: app/components/document/sidebar_component.html.erb [hunk 4]
```diff
@@ -1,17 +1,13 @@
...
-<%= render AlsoAvailableComponent.new(document:) %>
-
-<div class="mt-4">
-  <%= render Blacklight::Document::MoreLikeThisComponent.new(document:) %>
-</div>
\ No newline at end of file
+<% filtered_components = filter_components(secondary_components) %>
+<% if filtered_components.any? %>
+  <div class="bg-fog-light component-container rounded mt-4">
+    <%= filtered_components.join(separator).html_safe %>
```
Comment (jcoyne) [inline app/components/document/sidebar_component.html.erb:11]: Can we use safe_join here?
```suggestion
    <%= safe_join(filtered_components, separator) %>
```

File: app/components/document/sidebar_component.html.erb [hunk 5]
```diff
@@ -1,17 +1,13 @@
...
-<% if filtered_components.any? || tools.present?  %>
+<% filtered_components = filter_components(components) %>
+<% if filtered_components.any?  %>
   <div class="bg-fog-light component-container rounded">
-    <%= filtered_components.join('<hr style="border-top: dotted 1px;" />').html_safe if filtered_components.any? %>
-    <% if filtered_components.any? && tools.present? %>
-      <hr style="border-top: dotted 1px;" />
-    <% end %>
-    <%= tools %>
+    <%= filtered_components.join(separator).html_safe %>
```
Comment (jcoyne) [inline app/components/document/sidebar_component.html.erb:4]:
```suggestion
    <%= safe_join filtered_components, separator %>
```

**Comment by thatbudakguy** [comment]
Thanks for the tip re: `safe_join`; updated

**Review by jcoyne — APPROVED** [review]

**Pull request merged**